File: TichuTallyAndroid/app/src/main/java/com/tichutally/app/domain/model/Game.kt
package com.tichutally.app.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class Game(
    val teamA: Team = Team(TeamType.TEAM_A),
    val teamB: Team = Team(TeamType.TEAM_B),
    val rounds: List<Round> = emptyList(),
    val targetScore: Int = 1000
) {
    val winner: TeamType?
        get() = when {
            teamA.totalScore >= targetScore && teamA.totalScore > teamB.totalScore -> TeamType.TEAM_A
            teamB.totalScore >= targetScore && teamB.totalScore > teamA.totalScore -> TeamType.TEAM_B
            else -> null
        }

    val isGameOver: Boolean
        get() = winner != null
}

File: TPoint/TPoint/Domain/Entities/Game.swift
//
//  Game.swift
//  TPoint
//

import Foundation

struct Game: Codable {
    var teamA: Team
    var teamB: Team
    var rounds: [Round]
    var targetScore: Int

    init(targetScore: Int = 1000) {
        self.teamA = Team(type: .teamA)
        self.teamB = Team(type: .teamB)
        self.rounds = []
        self.targetScore = targetScore
    }

    var winner: TeamType? {
        if teamA.totalScore >= targetScore && teamA.totalScore > teamB.totalScore {
            return .teamA
        } else if teamB.totalScore >= targetScore && teamB.totalScore > teamA.totalScore {
            return .teamB
        }
        return nil
    }

    var isGameOver: Bool {
        winner != nil
    }

    mutating func reset() {
        teamA = Team(type: .teamA)
        teamB = Team(type: .teamB)
        rounds = []
    }
}
